capstone-backend/src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -0,0 +1,93 @@
+package com.google.sps.servlets;
+
+import com.google.sps.data.User;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.WriteResult;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that stores User's data from Frontend Authentication. */
+@WebServlet("/api/user")
+public class UserServlet extends HttpServlet {
+
+  private Firestore getFirestore() throws IOException {
+    return Utility.getFirestoreDb();
+  }
+
+  @Override
```

Comment:
It might be good to have a doGet as well to use for example, on a profile page where we can display the user's info and their friends

Reply:
Yeah that sounds good. Its probably outside the scope of this PR.